### PR TITLE
bpo-24638: Improve the error message in asyncio.ensure_future()

### PR DESCRIPTION
**Pull request by ZackerySpytz**
https://bugs.python.org/issue24638

### CODE REVIEW

File: Lib/asyncio/tasks.py [hunk 1]
```diff
@@ -627,7 +627,8 @@ def ensure_future(coro_or_future, *, loop=None):
         return task
     elif futures.isfuture(coro_or_future):
         if loop is not None and loop is not futures._get_loop(coro_or_future):
-            raise ValueError('loop argument must agree with Future')
+            raise ValueError('The future belongs to a different loop than '
```
Comment (eamanu) [inline Lib/asyncio/tasks.py:630]: What about "The future or coroutine"?

Comment (ZackerySpytz) [reply Lib/asyncio/tasks.py:630]: Coroutines are handled by the `if` just above, no?

Comment (eamanu) [reply Lib/asyncio/tasks.py:630]: hmmm I am not sure, but if I read the name `coro_or_future` I think that the ```_get_loop(coro_or_routine)``` handle both future and coroutine.

Comment (eamanu) [reply Lib/asyncio/tasks.py]: hmmm I am not sure, but if I read the name coro_or_future I think that the _get_loop(coro_or_routine) handle both future and coroutine.

**Comment by SpotlightKid** [comment]
I just stumbled over this non-helpful error message myself, because some code I was using, creates some `asyncio.Future`s using the default event loop, even though it allows to set the event loop in other places. The error message gave me no real clue as to the source of the problem, so I would appreciate this change.

File: Lib/asyncio/tasks.py (hunk 1 above)
Comment (ZackerySpytz) [reply Lib/asyncio/tasks.py:630]: Well, it seems that you're incorrect. :)

**Comment by ZackerySpytz** [comment]
@SpotlightKid Thank you for the support.

@1st1 @asvetlov Can you please take a look at this PR? Thanks.

**Review by asvetlov — APPROVED** [review]

**Pull request merged**